Assets/Scripts/NPCs/Patroller.cs

```diff
@@ -48,20 +40,33 @@ public void SetPoints(List<Vector3> newPoints)
         points = new List<Vector3>(newPoints);
     }
 
-    public void HandleTargetsInRange(int numTargetsInRange)
+    public override void HandleTargetsInRange(int numTargetsInRange)
     {
-        if (chase && numTargetsInRange > 0 && currentState != NurseStates.Chasing)
+        if (chase && numTargetsInRange > 0 && currentState != ActionStates.Chasing)
         {
             ChaseTarget();
-            currentState = NurseStates.Chasing;
+            currentState = ActionStates.Chasing;
         }
-        else if (numTargetsInRange == 0 && currentState != NurseStates.Patrolling)
+        else if (numTargetsInRange == 0 && currentState != ActionStates.Patrolling)
         {
             GotoNextPoint();
-            currentState = NurseStates.Patrolling;
+            currentState = ActionStates.Patrolling;
         }
     }
 
+    bool WaypointPauseComplete()
```

Comment:
Not needed for this pull request, but I'd keep in mind future updates to include `EnemyWaypoint.stayTime` from `MissionCafeteria1.cs` to have this wait time changeable, and per-waypoint.